### PR TITLE
Add line number information with Dwarf in elf executable

### PR DESCRIPTION
**Pull request by s123unny**

### CODE REVIEW

File: aeneas/src/os/Linux.v3 [hunk 1]
```diff
@@ -142,32 +143,40 @@ class LinuxTarget extends Target {
...
-				m.source.renderLong(buf);
-				buf.out(w.putkv);
-				w.putb(0);
+			if (CLOptions.SYMBOLS.val) {
+				sections.addFuncSymbol(w, ".entry", mach.entryStub);
+				if (mach.allocStub != null) sections.addFuncSymbol(w, ".alloc", mach.allocStub);
+				if (mach.signalStub != null) sections.addFuncSymbol(w, ".signal", mach.signalStub);
+
+				var buf = StringBuilder.new();
+				for (i < prog.ir.methods.length) {
```
Comment (titzer) [inline aeneas/src/os/Linux.v3:152]: Now that this is nested another level, it's probably a good idea for us to factor out a method `addMethodSymbol()`.

**Review by titzer — APPROVED** [review]
lgtm with minor suggestion

File: aeneas/src/os/Linux.v3 [hunk 2]
```diff
@@ -285,6 +263,31 @@ class ElfSections(elf: ElfHeader) {
...
 		str.sh_size = w.pos - str_start;
 	}
+	// Add symbol entry and write method name to .strtab section
+	def addMethodSymbol(w: MachDataWriter, mach: MachProgram, prog: Program) {
+		addFuncSymbol(w, ".entry", mach.entryStub);
+		if (mach.allocStub != null) addFuncSymbol(w, ".alloc", mach.allocStub);
+		if (mach.signalStub != null) addFuncSymbol(w, ".signal", mach.signalStub);
+
+		var buf = StringBuilder.new();
+		for (i < prog.ir.methods.length) {
```
Comment (titzer) [inline aeneas/src/os/Linux.v3:273]: Oh, I meant just to pull out the body of the loop that works on a single method.

Comment (s123unny) [reply aeneas/src/os/Linux.v3:273]: What about the stringBuilder? or use `addFuncSymbol()` as well?

Comment (titzer) [reply aeneas/src/os/Linux.v3:273]: Yeah, you can pass in the StringBuilder as a temporary. I didn't quite understand the `addFuncSymbol()` comment. If you go back to before extracting this method and only extract the loop body, I think that solves that.

**Review by titzer — APPROVED** [review]
lgtm, thanks!

**Pull request merged**